InputValidation/IValidation.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+/// <summary>
+/// Interface for various forms of input validation in TotalImage.
+/// </summary>
+namespace TotalImage.Validation
+{
+    public interface IValidation
+    {
+        object Validate(object Data); // Can return modified data. For example, upper-to-lower case.
```

Comment:
If we go with a generic interface, the parameter should then be `T Data`

However, it looks like you're indicating invalid data by returning a `null`, I'm not sure if that's best. It's quite possible that valid data will sometimes be a `null`, so perhaps return a ValidationResult<T> object, that has a `bool` Property `IsValid` and then a `T` property called `Result` with the current return data?